src/picongpu/include/fields/laserProfiles/laserGaussianBeam.hpp

```diff
@@ -122,9 +122,10 @@ namespace picongpu
 
             // initialize temporary variables
             uint32_t m = 0;
-            float_X etrans = 0.0f;
-            float_X etrans_norm = 0.0f;
-            for ( m=0; m<MODENUMBER ; m++ ) etrans_norm += LAGUERREMODES[m];
+            float_X etrans = float_X(0.0);
+            float_X etrans_norm = float_X(0.0);
+            PMACC_CASSERT_MSG(MODENUMBER_must_be_smaller_than_number_of_entries_in_LAGUERREMODES_vector, MODENUMBER<LAGUERREMODES_t::dim);
```

Comment:
Since the math function does not directly work with the mode-vector, I included the compile-assertion in the laser implementation before first use.